File: test/data/phenology/crop2ml/algo/cs/updateCalendar.cs

if ((phase >= 1.0 && phase < 2.0) && (calendarMoments.Contains("Emergence")==false ))
{
    calendarMoments.Add("Emergence");
    calendarCumuls.Add(cumulTT);
    calendarDates.Add(currentdate);
}
else if ((phase >= 2.0 && phase < 3.0)  && ( calendarMoments.Contains("FloralInitiation")==false  ))
{
    calendarMoments.Add("FloralInitiation") ;
    calendarCumuls.Add(cumulTT);
    calendarDates.Add(currentdate);
}
else if ((phase >= 3.0 && phase < 4.0)  && (calendarMoments.Contains("Heading")==false  ))
{
    calendarMoments.Add("Heading");
    calendarCumuls.Add(cumulTT);
    calendarDates.Add(currentdate);
}
else if ((phase == 4.0)  && (calendarMoments.Contains("Anthesis" )==false  ))
{
    calendarMoments.Add("Anthesis");
    calendarCumuls.Add(cumulTT);
    calendarDates.Add(currentdate);
}
else if ((phase == 4.5)  && (calendarMoments.Contains("EndCellDivision" )==false  ))
{
    calendarMoments.Add("EndCellDivision");
    calendarCumuls.Add(cumulTT);
    calendarDates.Add(currentdate);
}
else if ((phase >= 5.0 && phase < 6.0) && ( calendarMoments.Contains("EndGrainFilling")==false  ))
{
    calendarMoments.Add("EndGrainFilling");
    calendarCumuls.Add(cumulTT);
    calendarDates.Add(currentdate);
}
else if ((phase >= 6.0 && phase < 7.0)  && (calendarMoments.Contains("Maturity" )==false ))
{
    calendarMoments.Add("Maturity");
    calendarCumuls.Add(cumulTT);
    calendarDates.Add(currentdate);
}

File: test/data/phenology/crop2ml/algo/cpp/updateCalendar.cpp

size_t pos = find(calendarMoments.begin(), calendarMoments.end(),"Emergence" ) - calendarMoments.begin();
if ((phase >= 1 && phase < 2) && (pos>=calendarMoments.size()))
{
    calendarMoments.push_back("Emergence");
    calendarCumuls.push_back(cumulTT);
    calendarDates.push_back(currentdate);
}
else if ((phase >= 2 && phase < 3)  && (size_t(find(calendarMoments.begin(), calendarMoments.end(),"FloralInitiation" ) - calendarMoments.begin()))>=calendarMoments.size())
{
    calendarMoments.push_back("FloralInitiation") ;
    calendarCumuls.push_back(cumulTT);
    calendarDates.push_back(currentdate);
}
else if ((phase >= 3 && phase < 4)  && (size_t(find(calendarMoments.begin(), calendarMoments.end(),"Heading" ) - calendarMoments.begin()))>=calendarMoments.size())
{
    calendarMoments.push_back("Heading");
    calendarCumuls.push_back(cumulTT);
    calendarDates.push_back(currentdate);
}
else if ((phase == 4)  && (size_t(find(calendarMoments.begin(), calendarMoments.end(),"Anthesis" ) - calendarMoments.begin()))>=calendarMoments.size())
{
    calendarMoments.push_back("Anthesis");
    calendarCumuls.push_back(cumulTT);
    calendarDates.push_back(currentdate);
}
else if ((phase == 4.5)  &&(size_t(find(calendarMoments.begin(), calendarMoments.end(),"EndCellDivision" ) - calendarMoments.begin()))>=calendarMoments.size())
{
    calendarMoments.push_back("EndCellDivision");
    calendarCumuls.push_back(cumulTT);
    calendarDates.push_back(currentdate);
}
else if ((phase >= 5 && phase < 6) && (size_t(find(calendarMoments.begin(), calendarMoments.end(),"EndGrainFilling" ) - calendarMoments.begin()))>=calendarMoments.size())
{
    calendarMoments.push_back("EndGrainFilling");
    calendarCumuls.push_back(cumulTT);
    calendarDates.push_back(currentdate);
}
else if ((phase >= 6 && phase < 7)  && (size_t(find(calendarMoments.begin(), calendarMoments.end(),"Maturity" ) - calendarMoments.begin()))>=calendarMoments.size())
{
    calendarMoments.push_back("Maturity");
    calendarCumuls.push_back(cumulTT);
    calendarDates.push_back(currentdate);
}




